connexion/apis/starlette_api.py

```diff
@@ -0,0 +1,333 @@
+            status_code=200,
+            media_type="text/yaml",
+        )
+
+    def add_swagger_ui(self):
+        """
+        Adds swagger ui to {base_path}/ui/
+        """
+        console_ui_path = self.options.openapi_console_ui_path.strip().rstrip("/")
+        logger.debug("Adding swagger-ui: %s%s/", self.base_path, console_ui_path)
+
+        for path in (
+            console_ui_path + "/",
+            console_ui_path + "/index.html",
+        ):
+            self.subapp.add_route(
+                methods=["GET"], path=path, endpoint=self._get_swagger_ui_home
+            )
+
+        if self.options.openapi_console_ui_config is not None:
+            self.subapp.add_route(
+                methods=["GET"],
+                path=console_ui_path + "/swagger-ui-config.json",
+                endpoint=self._get_swagger_ui_config,
+            )
+
+        # we have to add an explicit redirect instead of relying on the
+        # normalize_path_middleware because we also serve static files
+        # from this dir (below)
+
+        async def redirect(request):
+            return RedirectResponse(url=self.base_path + console_ui_path + "/")
+
+        self.subapp.add_route(methods=["GET"], path=console_ui_path, endpoint=redirect)
+
+        # this route will match and get a permission error when trying to
+        # serve index.html, so we add the redirect above.
+        self.subapp.mount(
+            path=console_ui_path,
+            app=StaticFiles(directory=str(self.options.openapi_console_ui_from_dir)),
+            name="swagger_ui_static",
+        )
+
+    async def _get_swagger_ui_home(self, req):
+        base_path = self._base_path_for_prefix(req)
+        template_variables = {
+            "request": req,
+            "openapi_spec_url": (base_path + self.options.openapi_spec_path),
+            **self.options.openapi_console_ui_index_template_variables,
+        }
+        if self.options.openapi_console_ui_config is not None:
+            template_variables["configUrl"] = "swagger-ui-config.json"
+
+        return self._templates.TemplateResponse("index.j2", template_variables)
+
+    async def _get_swagger_ui_config(self, request):
+        return Response(
+            status_code=200,
+            media_type="text/json",
+            content=self.jsonifier.dumps(self.options.openapi_console_ui_config),
+        )
+
+    def add_auth_on_not_found(self, security, security_definitions):
+        """
+        Adds a 404 error handler to authenticate and only expose the 404 status if the security validation pass.
+        """
+        logger.debug("Adding path not found authentication")
+        not_found_error = AuthErrorHandler(
+            self,
+            NotFound(),
+            security=security,
+            security_definitions=security_definitions,
+        )
+        endpoint_name = f"{self._api_name}_not_found"
+        self.subapp.add_route(
+            path="/{not_found_path}",
```

Comment:
What is the intention here? Shouldn't we be using [exception handlers](https://www.starlette.io/exceptions/)?

Reply:
I tried to diverge as little as possible from the aiohttp implementation, but this can be changed. Will look at it tomorrow